src/lava/magma/core/run_configs.py

```diff
@@ -70,57 +70,134 @@ def select(self,
         super().__init__(custom_sync_domains=custom_sync_domains)
         self.select_tag = select_tag
         self.select_sub_proc_model = select_sub_proc_model
         self.exception_proc_model_map = exception_proc_model_map
         if not exception_proc_model_map:
             self.exception_proc_model_map = {}
 
-    def select(self, proc, proc_models):
+    def select(self,
+               proc: AbstractProcess,
+               proc_models: ty.List[ty.Type[AbstractProcessModel]]) \
+            -> ty.Type[AbstractProcessModel]:
+        """
+        Selects an appropriate ProcessModel from a list of ProcessModels for
+        a Process, based on user requests
+        Parameters
+        ----------
+        proc: AbstractProcess, Process for which ProcessModel is selected
+        proc_models: List[AbstractProcessModel], list of ProcessModels of
+            Process
+
+        Returns
+        -------
+        Selected ProcessModel class
+        """
         from lava.magma.core.model.sub.model import AbstractSubProcessModel
         from lava.magma.core.model.py.model import AbstractPyProcessModel
+
+        def issubpm(pm: ty.Type[AbstractProcessModel]) -> bool:
+            """Checks if input ProcessModel is a SubProcessModel"""
+            return issubclass(pm, AbstractSubProcessModel)
+
+        def ispypm(pm: ty.Type[AbstractProcessModel]) -> bool:
+            """Checks if input ProcessModel is a PyProcessModel"""
+            return issubclass(pm, AbstractPyProcessModel)
+
         # First priority to the exceptions. We will simply return the
         # ProcessModel class associated with a Process class in the
         # exceptions dictionary
         if proc.__class__ in self.exception_proc_model_map:
             return self.exception_proc_model_map[proc.__class__]
-        # Now, we loop over all ProcessModels proc_models of a Process proc
-        for pm in proc_models:
-            # Priority given to SubProcessModels
-            if issubclass(pm, AbstractSubProcessModel) and \
-                    self.select_sub_proc_model:
+
+        # If there is only one ProcessModel available, then we just return
+        # it by default after confirming that it is Sub or Py ProcessModel
+        if len(proc_models) == 1 and (issubpm(proc_models[0]) or ispypm(
+                proc_models[0])):
+            print(f"[{self.__class__.__qualname__}]: Using the only "
+                  f"ProcessModel {proc_models[0].__qualname__} "
+                  f"available for Process "
+                  f"{proc.name}::{proc.__class__.__qualname__}.")
+            return proc_models[0]
+
+        # If SubProcessModel is asked for
+        if self.select_sub_proc_model:
+            # Get the indices of SubProcessModels in the list of ProcessModels
+            sub_pm_idxs = [idx for idx, pm in enumerate(proc_models) if
+                           issubpm(pm)]
+            if len(sub_pm_idxs) == 0:
```

Comment:
No this won't work in general. select_sub_proc_model can only express a priority/preference not a strict requirement. At one point the compiler will have reached the bottom in its search for a ProcModel. If that Process has more than one LeafProcessModel then the prior "if" won't catch it and then this throws an error because there's no infinite continuation for the sub-Process decomposition.

Reply:
This is a review of stale code for some reason. Already caught this and changed it.